stdlib/public/core/Codable.swift

```diff
@@ -3635,208 +3633,208 @@ internal final class _KeyedEncodingContainerBox<
     return concrete.codingPath
   }
 
-  override internal func encodeNil(forKey key: Key) throws {
-    try concrete.encodeNil(forKey: key)
+  override internal func encodeNil(forKey key: CodingKey) throws {
+    try concrete.encodeNil(forKey: key as! Key)
   }
 
-  override internal func encode(_ value: Bool, forKey key: Key) throws {
-    try concrete.encode(value, forKey: key)
+  override internal func encode(_ value: Bool, forKey key: CodingKey) throws {
```

Comment:
Instead of using `CodingKey` existentials, it should be cheaper to pass a generic `T: CodingKey` argument in. Then, instead of `key as! Key`, we could also use `assert(Key.self == T.self); unsafeBitCast(key, to: Key.self)` since we know the type should always match.

Reply:
Good idea.  That won't spit out any additional metadata?

Reply:
If we don't specialize, it'll be the same as the existential form metadata wise, but we're more likely to specialize a generic argument, which in most cases should eliminate the need for metadata.